tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+            permission=True,
+            name="Door1",
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.CLOSED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=2,
+            temperature=None,
+        ),
+        door2=Door(
+            door_id=2,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        door3=Door(
+            door_id=3,
+            permission=True,
+            name=None,
+            mode=DoorMode.GARAGE,
+            status=DoorStatus.UNDEFINED,
+            sensor=True,
+            sensorid=None,
+            camera=False,
+            events=0,
+            temperature=None,
+        ),
+        outputs=Outputs(output1=True, output2=False, output3=True),
+        network=Network(ip=""),
+        wifi=Wifi(SSID="", linkquality="", signal=""),
+    )
+
+    def new_api(ip_address: str, username: str, password: str) -> GogoGate2Api:
+        if ip_address == "127.0.1.0":
+            return api0
+        if ip_address == "127.0.1.1":
+            return api1
+        raise Exception(f"Untested ip address {ip_address}")
+
+    component_factory.api_class_mock.side_effect = new_api
+
+    await component_factory.configure_component(
+        cover_config=[
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover0",
+                CONF_IP_ADDRESS: "127.0.1.0",
+                CONF_USERNAME: "user0",
+                CONF_PASSWORD: "password0",
+            },
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover1",
+                CONF_IP_ADDRESS: "127.0.1.1",
+                CONF_USERNAME: "user1",
+                CONF_PASSWORD: "password1",
+            },
+        ]
+    )
+    entity_ids = hass.states.async_entity_ids(COVER_DOMAIN)
+    assert entity_ids is not None
+    assert len(entity_ids) == 2
+    assert "cover.door1" in entity_ids
+    assert "cover.door1_2" in entity_ids
+
+    await component_factory.unload()
+
+
+async def test_cover_update(
+    hass: HomeAssistant, component_factory: ComponentFactory
+) -> None:
+    """Test cover."""
+    await component_factory.configure_component()
+    component_data = await component_factory.run_config_flow(
+        config_data={
+            CONF_IP_ADDRESS: "127.0.0.2",
+            CONF_USERNAME: "user0",
+            CONF_PASSWORD: "password0",
+        }
+    )
+
+    assert hass.states.async_entity_ids(COVER_DOMAIN)
+
+    state = hass.states.get("cover.door1")
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes["friendly_name"] == "Door1"
+    assert state.attributes["supported_features"] == 3
+    assert state.attributes["device_class"] == "garage"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)